drivers/adc/adc_max1125x.c

```diff
@@ -0,0 +1,811 @@
+	const struct spi_buf_set rx = {.buffers = &rx_buf, .count = 1};
+
+	gpio_pin_set_dt(&config->csb_gpio, 1);
+	ret = spi_write_dt(&config->bus, &tx);
+	if (ret != 0) {
+		LOG_ERR("MAX1125X: error writing register 0x%X (%d)", reg_addr, ret);
+		return ret;
+	}
+	ret = spi_read_dt(&config->bus, &rx);
+	if (ret != 0) {
+		LOG_ERR("MAX1125X: error reading register 0x%X (%d)", reg_addr, ret);
+		return ret;
+	}
+	gpio_pin_set_dt(&config->csb_gpio, 0);
+
+	return 0;
+}
+
+static int max1125x_write_reg(const struct device *dev, enum max1125x_reg reg_addr,
+			      uint8_t *reg_val, size_t reg_size)
+{
+	int ret;
+	const struct max1125x_config *config = dev->config;
+	uint8_t command = MAX1125X_CMD_WRITE | reg_addr;
+
+	const struct spi_buf spi_buf[2] = {{.buf = &command, .len = sizeof(command)},
+					   {.buf = reg_val, .len = reg_size}};
+	const struct spi_buf_set tx = {.buffers = spi_buf, .count = ARRAY_SIZE(spi_buf)};
+
+	gpio_pin_set_dt(&config->csb_gpio, 1);
+	ret = spi_write_dt(&config->bus, &tx);
+	if (ret != 0) {
+		LOG_ERR("MAX1125X: error writing register 0x%X (%d)", reg_addr, ret);
+		return ret;
+	}
+	gpio_pin_set_dt(&config->csb_gpio, 0);
+
+	return 0;
+}
+
+static int max1125x_send_command(const struct device *dev, enum max1125x_mode mode, uint8_t rate)
+{
+	int ret;
+	const struct max1125x_config *config = dev->config;
+	uint8_t command = MAX1125X_CMD_CONV | mode | rate;
+	const struct spi_buf spi_buf = {.buf = &command, .len = sizeof(command)};
+	const struct spi_buf_set tx = {.buffers = &spi_buf, .count = 1};
+
+	gpio_pin_set_dt(&config->csb_gpio, 1);
+	ret = spi_write_dt(&config->bus, &tx);
+	if (ret != 0) {
+		LOG_ERR("MAX1125X: error writing register 0x%X (%d)", rate, ret);
+		return ret;
+	}
+	gpio_pin_set_dt(&config->csb_gpio, 0);
+
+	return 0;
+}
+
+static int max1125x_start_conversion(const struct device *dev)
+{
+	const struct max1125x_data *data = dev->data;
+
+	return max1125x_send_command(dev, MAX1125X_CMD_SEQUENCER, data->rate);
+}
+
+static inline int max1125x_acq_time_to_dr(const struct device *dev, uint16_t acq_time)
+{
+	struct max1125x_data *data = dev->data;
+	const struct max1125x_config *config = dev->config;
+	const uint32_t *odr_delay = config->odr_delay;
+	uint32_t odr_delay_us = 0;
+	uint16_t acq_value = ADC_ACQ_TIME_VALUE(acq_time);
+	int odr = -EINVAL;
+
+	if (acq_time != ADC_ACQ_TIME_DEFAULT && ADC_ACQ_TIME_UNIT(acq_time) != ADC_ACQ_TIME_TICKS) {
+		LOG_ERR("MAX1125X: invalid acq time value (%d)", acq_time);
+		return -EINVAL;
+	}
+
+	if (acq_value < MAX1125X_CONFIG_RATE_1_9 || acq_value > MAX1125X_CONFIG_RATE_64000) {
+		LOG_ERR("MAX1125X: invalid acq value (%d)", acq_value);
+		return -EINVAL;
+	}
+
+	odr = acq_value;
+	odr_delay_us = odr_delay[acq_value];
+
+	data->rate = odr;
+	data->ready_time = K_USEC(odr_delay_us);
+
+	return odr;
+}
+
+static int max1125x_wait_data_ready(const struct device *dev)
+{
+	struct max1125x_data *data = dev->data;
+
+	k_sleep(data->ready_time);
+
+	uint8_t is_ready
```

Comment:
```suggestion
	uint8_t is_ready;
```
This leads me to believe this driver isn't getting built in CI. Have you tried building `tests/drivers/build_all/adc` locally?

Reply:
yes i run tested before pushed.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/58497719/92085323-f436-4a95-9f5c-06d705a0b144)